app/controllers/bookmarks_controller.rb

```diff
@@ -132,23 +132,22 @@ def status_action documents
     status = params['action']
     Array(documents.map(&:id)).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
+      if status == 'publish'
+        success_ids << id if can?(:update, media_object)
```

Comment:
The assumption here is that the person might have selected some MediaObjects they can update, some they can't. For the ones they can change, they will update (publish/unpublish) and show a notice via flash, the ones that can't will have the error reported via flash. (This is logic that pre-dates this patch, btw.). I hope I've understood your question.